### PR TITLE
Add subcommand to visualize the requests running concurrently using a trace file

### PR DESCRIPTION
**Pull request by ahoppen**
Showing the list of active requests didn’t turn out to be very useful for debugging. What is more useful, however, is to visualize the requests as the get enqueued and handled in a trace chart. This gives a very intuitive view of which request is blocking another request from getting executed.

For example, I was able to generate the following trace, which immediately shows that document requests are blocked from executing until the `WorkspaceTestsRequest` finishes (an issue that we should fix). 

Screenshot 2024-09-27 at 22 30 34

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/Diagnose/TraceFromSignpostsCommand.swift [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+//===----------------------------------------------------------------------===//
+
+import ArgumentParser
+import Foundation
+import RegexBuilder
+
+import class TSCBasic.Process
+
+/// Shared instance of the regex that is used to extract Signpost lines from `log stream --signpost`.
+struct LogParseRegex {
```
Comment (hamishknight) [inline Sources/Diagnose/TraceFromSignpostsCommand.swift:20]: Should this be fileprivate?

File: Sources/Diagnose/TraceFromSignpostsCommand.swift [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+  private(set) var regex:
+    Regex<Regex<(Substring, Substring, Substring, Substring, Substring, Substring, Substring, Substring)>.RegexOutput>!
+
+  private init() {
+    regex = Regex {
+      Capture(as: dateComponent) {
+        #/[-0-9]+ [0-9:.-]+/#
+      }
+      " "
+      #/[0-9abcdefx]+/#  // Thread ID
```
Comment (hamishknight) [inline Sources/Diagnose/TraceFromSignpostsCommand.swift:38]: FWIW you can do `a-f`, doesn't really matter though

File: Sources/Diagnose/TraceFromSignpostsCommand.swift [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+        OneOrMore(.hexDigit)
+      }
+      ", process, "
+      ZeroOrMore(.whitespace)
+      Capture(as: eventTypeComponent) {
+        #/(begin|event|end)/#
+      }
+      "]"
+      ZeroOrMore(.whitespace)
+      #/[^ ]+/#  // Process name
```
Comment (hamishknight) [inline Sources/Diagnose/TraceFromSignpostsCommand.swift:61]: Could be `OneOrMore(.whitespace.inverted)`, doesn't really matter though

**Review by hamishknight — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**